packages/grid/x-grid/src/useXGridComponent.tsx

```diff
@@ -4,6 +4,7 @@ import { useGridColumnMenu } from '../../_modules_/grid/hooks/features/columnMen
 import { useGridColumnReorder } from '../../_modules_/grid/hooks/features/columnReorder/useGridColumnReorder';
 import { useGridColumnResize } from '../../_modules_/grid/hooks/features/columnResize/useGridColumnResize';
 import { useGridColumns } from '../../_modules_/grid/hooks/features/columns/useGridColumns';
+import { useGridVisibleColumns } from '../../_modules_/grid/hooks/features/columns/useGridVisibleColumns';
```

Comment:
How about we only allow one top-level hook feature per folder? Either the feature can be used on its own and

```suggestion
import { useGridVisibleColumns } from '../../_modules_/grid/hooks/features/columnVisibility/useGridVisibleColumns';
```

or it can't it should already be part of `useGridColumns.ts`

```suggestion
```

Reply:
You mean be a part of `useGridColumns.ts` like `usGrideVirtualColumns` is called from `useGridVirtualRows` ?